libucontext.yaml

```diff
@@ -15,7 +15,10 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
+      - openssf-compiler-options
       - scdoc
+  environment:
+    CFLAGS: "-Wno-incompatible-pointer-types"
```

Comment:
I verified locally that this does not have any effect.  Can you drop it?
(Please feel free to re-verify that)

Reply:
That does not seem to be the case for me:

```
2024/09/24 07:53:10 WARN arch/aarch64/makecontext.c:48:14: error: assignment to 'long unsigned int *' from incompatible pointer type 'long long unsigned int *' [-Wincompatible-pointer-types]
2024/09/24 07:53:10 WARN    48 |         regp = &(ucp->uc_mcontext.regs[0]);
2024/09/24 07:53:10 WARN       |              ^
2024/09/24 07:53:10 WARN make: *** [Makefile:130: arch/aarch64/makecontext.o] Error 1
2024/09/24 07:53:10 WARN make: *** Waiting for unfinished jobs....
2024/09/24 07:53:10 INFO make: Leaving directory '/home/build'
2024/09/24 07:53:11 INFO pod 5011032eebf833f58c74e6c677694b1ed3d032ade4f713a64d339c6d4196b65c terminated
2024/09/24 07:53:11 ERRO ERROR: failed to build package. the build environment has been preserved:
2024/09/24 07:53:11 INFO   workspace dir: /var/folders/vw/5gq4f3g53y55wkv1phvjx5b00000gn/T/melange-workspace-3982346288
2024/09/24 07:53:11 INFO   guest dir: /var/folders/vw/5gq4f3g53y55wkv1phvjx5b00000gn/T/melange-guest-3270809625
2024/09/24 07:53:11 ERRO failed to build package: unable to run package libucontext pipeline: unable to run pipeline: unable to run pipeline: task exited with code 2
make[1]: *** [packages/aarch64/libucontext-1.2-r2.apk] Error 1
make: *** [package/libucontext] Error 2

~/src/os jvreeland/fix-libucontext* ≡
❯ git diff
diff --git a/libucontext.yaml b/libucontext.yaml
index 1844794e6..1e02a9873 100644
--- a/libucontext.yaml
+++ b/libucontext.yaml
@@ -17,8 +17,6 @@ environment:
       - ca-certificates-bundle
       - openssf-compiler-options
       - scdoc
-  environment:
-    CFLAGS: "-Wno-incompatible-pointer-types"
 
 pipeline:
   - uses: git-checkout
```

Reply:
I put up https://github.com/wolfi-dev/os/pull/29242 . We'll see if c-i agrees with me or you.

Reply:
can you post a full build log of your failing ?

Reply:
I can looks like CI failed though:

```
2024/09/24 15:11:38 WARN /usr/include/features.h:197:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
2024/09/24 15:11:38 WARN   197 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
2024/09/24 15:11:38 WARN       |   ^~~~~~~
2024/09/24 15:11:38 WARN arch/aarch64/makecontext.c: In function 'libucontext_makecontext':
2024/09/24 15:11:38 WARN arch/aarch64/makecontext.c:48:14: error: assignment to 'long unsigned int *' from incompatible pointer type 'long long unsigned int *' [-Wincompatible-pointer-types]
2024/09/24 15:11:38 WARN    48 |         regp = &(ucp->uc_mcontext.regs[0]);
2024/09/24 15:11:38 WARN       |              ^
2024/09/24 15:11:38 WARN make: *** [Makefile:130: arch/aarch64/makecontext.o] Error 1
2024/09/24 15:11:38 WARN make: *** Waiting for unfinished jobs....
2024/09/24 15:11:38 INFO make: Leaving directory '/home/build'
2024/09/24 15:11:38 ERRO ERROR: failed to build package. the build environment has been preserved:
2024/09/24 15:11:38 INFO   workspace dir: /temp/melange-workspace-2906867
2024/09/24 15:11:38 INFO   guest dir: /temp/melange-guest-4006211376
2024/09/24 15:11:38 ERRO failed to build package: unable to run package libucontext pipeline: unable to run pipeline: unable to run pipeline: exit status 2
make[1]: *** [Makefile:111: packages/aarch64/libucontext-1.2-r2.apk] Error 1

```

Reply:
ok. good news.  it failed for c-i.
so its actually arch specific. and i was building x86_64 and you were running aarch64. so that makes sense.